src/arithmetic/boolean_funcs/boolean_funcs.c

```diff
@@ -75,7 +76,8 @@ SIValue AR_GT(SIValue *argv, int argc) {
 		char *error;
 		asprintf(&error, "Type mismatch: expected %s but was %s", SIType_ToString(SI_TYPE(a)),
 				 SIType_ToString(SI_TYPE(b)));
-		return SI_Error(error);
+		QueryCtx_SetError(error);
```

Comment:
In which situation does `a` and `b` are not comparable?
at this point we're guarantee that `a` and `b` are of type: `SI_NUMERIC, T_STRING, T_BOOL`
can't we say a numeric is greater than boolean? or whatever order Cypher dictates?

Reply:
The query `RETURN 'a' > 2` returns in Neo:
> Neo.ClientError.Statement.SyntaxError: Type mismatch: expected String but was Integer (line 1, column 14 (offset: 13))

Orderability and comparability are related but not identical concepts. Here, for example, we order non-comparable types:
`UNWIND [NULL, 5,'f',3, timestamp()] AS a RETURN a ORDER BY a`
```
1) 1) "a"
2) 1) 1) "f"
   2) 1) (integer) 3
   3) 1) (integer) 5
   4) 1) (integer) 1567867171834
   5) 1) (nil)
```
The two integers are comparable, and ordered accordingly, everything else is ordered by type precedence.